builtin/providers/aws/resource_aws_route53_record.go

```diff
@@ -202,9 +203,12 @@ func resourceAwsRoute53RecordDelete(d *schema.ResourceData, meta interface{}) er
 	zone := d.Get("zone_id").(string)
 	log.Printf("[DEBUG] Deleting resource records for zone: %s, name: %s",
 		zone, d.Get("name").(string))
-
+	zoneRecord, err := conn.GetHostedZone(&route53.GetHostedZoneRequest{ID: aws.String(zone)})
+	if err != nil {
+		return err
+	}
 	// Get the records
-	rec, err := resourceAwsRoute53RecordBuildSet(d)
+	rec, err := resourceAwsRoute53RecordBuildSet(d, *zoneRecord.HostedZone.Name)
```

Comment:
I don't _believe_ so; `name` is required to create the Zone, so, I'm trusting AWS to have that. Assuming so, then the `err`  check immediately before this should bail us out if the Zone isn't found. 

If you feel strongly enough I can add them though; generally I agree with your caution here.